### PR TITLE
LPD-37515 Add "scope" field to the "ThemeCSSCET" and validate it at build time

### PR DESCRIPTION
**Pull request by thiago-buarqque**
Coming from https://github.com/liferay-platform-experience/liferay-portal/pull/765

-----

Ticket: https://liferay.atlassian.net/browse/LPD-37515

# Goal

To provide a way of scoping ThemeCSSCET to control panel pages.

# Approach

As in `GlobalJSCET`, add a field named "scope". For the `ThemeCSSCET`, the valid values for `scope` are `controlPanel` and `page`.

-------

> If we have `GlobalJSCET` and `ThemeCSSCET` and, in the future, `GlobalCSSCET`, with the scope attribute, why don't you create a common interface and make them extends it?

I tried that. The issue is that we have a [SF rule](https://github.com/liferay/liferay-portal/blob/8ddfa14f02c01252eba38c94b1c1900895081132/modules/util/source-formatter/src/main/java/com/liferay/source/formatter/processor/CETSourceProcessor.java#L21) that updates [this file](https://github.com/liferay/liferay-portal/blob/3fcc602493fa10ec256599e4551b186b3bd233b5/modules/apps/client-extension/client-extension-type-api/src/main/resources/com/liferay/client/extension/type/dependencies/client-extension-types.json#L1) to match the interface of each type. That rule does not account for attributes coming from other interfaces that aren't the CX type itself.

### CODE REVIEW

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:gradle-plugins

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:gradle-plugins

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:report:53890311

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:gradle-plugins

File: modules/sdk/gradle-plugins-workspace/src/main/resources/schemas/client-extension.schema.json [hunk 1]
```diff
@@ -403,6 +403,9 @@
 								},
 								"mainURL": {
 									"type": "string"
+								},
+								"scope": {
+									"type": "string"
```
[the remark this answers is not in the document]
Comment (thiago-buarqque) [reply modules/sdk/gradle-plugins-workspace/src/main/resources/schemas/client-extension.schema.json]: Done, thanks @drewbrokke , I forgot about enums :sweat_smile: 

I took the opportunity to change a similar case for the GlobalJS CX

[the remark this answers is not in the document]
Comment (drewbrokke) [reply modules/sdk/gradle-plugins-workspace/src/main/resources/schemas/client-extension.schema.json]: Great, thanks @thiago-buarqque!

**Comment by thiago-buarqque** [comment]
@drewbrokke unrelated CI failures, may I forward this one?

**Review by drewbrokke — APPROVED** [review]
Looks good, thanks @thiago-buarqque 👍

**Comment by thiago-buarqque** [comment]
ci:forward:force

**Pull request closed without merging**